index.d.ts

```diff
@@ -20,24 +20,36 @@ interface BodyEntry {
 }
 
 export interface MultipartFields {
-    [x: string]: Multipart | Multipart[];
+    [fieldname: string]: Multipart | Multipart[];
 }
 
-export type Multipart<T = true> = T extends true ? MultipartFile : MultipartValue<T>;
+export type Multipart = MultipartFile | MultipartValue;
 
 export interface MultipartFile {
   toBuffer: () => Promise<Buffer>,
   file: BusboyFileStream,
-  filepath: string,
   fieldname: string,
   filename: string,
   encoding: string,
   mimetype: string,
   fields: MultipartFields
 }
 
-export interface MultipartValue<T> {
-  value: T
+export interface SavedMultipartFile extends MultipartFile {
+  /**
+   * Path to the temporary file
+   */
+  filepath: string,
+}
+
+export interface MultipartValue<T = unknown> {
+  value: T;
+  fieldname: string;
+  mimetype?: string;
```

Comment:
`string | undefined` because mime type get swallowed when we parse the value as JSON. I think this is a bug, but that's for another PR.